### PR TITLE
BUG: Fixes to several filters as reported by users

### PR DESCRIPTION
**Pull request by imikejackson**
WriteSTLFile
SampleTriangleGeometryOntoRegularGrid.

### CODE REVIEW

File: src/complex/Utilities/SampleSurfaceMesh.cpp [hunk 1]
```diff
@@ -306,21 +315,20 @@ void SampleSurfaceMesh::sendThreadSafeProgressMessage(usize featureId, usize num
...
+    std::string progMessage = fmt::format("{}", m_ProgressCounter);
+    //    float inverseRate = static_cast<float>(diff) / static_cast<float>(m_ProgressCounter - m_LastProgressInt);
+    //    auto remainMillis = std::chrono::milliseconds(static_cast<int64>(inverseRate * (totalFeatures - m_ProgressCounter)));
+    //    auto secs = std::chrono::duration_cast<std::chrono::seconds>(remainMillis);
+    //    remainMillis -= std::chrono::duration_cast<std::chrono::milliseconds>(secs);
+    //    auto mins = std::chrono::duration_cast<std::chrono::minutes>(secs);
+    //    secs -= std::chrono::duration_cast<std::chrono::seconds>(mins);
+    //    auto hour = std::chrono::duration_cast<std::chrono::hours>(mins);
+    //    mins -= std::chrono::duration_cast<std::chrono::minutes>(hour);
+    //    progMessage += fmt::format(" || Est. Time Remain: {} hours {} minutes {} seconds", hour.count(), mins.count(), secs.count());
```
Comment (nyoungbq) [inline src/complex/Utilities/SampleSurfaceMesh.cpp:329]:
```suggestion
```

File: src/complex/Utilities/SampleSurfaceMesh.cpp [hunk 2]
```diff
@@ -306,21 +315,20 @@ void SampleSurfaceMesh::sendThreadSafeProgressMessage(usize featureId, usize num
...
     m_MessageHandler({IFilter::Message::Type::Info, progMessage});
-    m_InitialTime = std::chrono::steady_clock::now();
-    m_LastProgressInt = m_ProgressCounter;
+    m_LastUpdateTime = std::chrono::steady_clock::now();
+    //    m_LastProgressInt = m_ProgressCounter;
```
Comment (nyoungbq) [inline src/complex/Utilities/SampleSurfaceMesh.cpp:332]:
```suggestion
```

File: src/complex/Utilities/Math/GeometryMath.hpp [hunk 3]
```diff
@@ -570,9 +580,20 @@ char IsPointInPolyhedron(const complex::TriangleGeom& faces, const std::vector<i
       }
       else
       {
-        std::array<Point3D<T>, 3> coords;
-        faces.getFaceCoordinates(faceIds[face], coords);
-        code = RayIntersectsTriangle(ray, coords[0], coords[1], coords[2]);
+        //        std::array<Point3D<T>, 3> coords;
+        //        triangleGeom.getFaceCoordinates(faceIds[face], coords);
```
Comment (nyoungbq) [inline src/complex/Utilities/Math/GeometryMath.hpp:584]:
```suggestion
```

**Review by nyoungbq — APPROVED** [review]

**Comment by imikejackson** [comment]
Closing as these changes are going to be on another PR or 2.

**Pull request closed without merging**